### DISCUSSION TITLE
SQLModel unknown type as prop on BaseModel

### DISCUSSION THREAD

**Discussion by EvanSchalton** [fastapi/sqlmodel#822] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel
from pydantic import BaseModel

class Table(SQLModel, table=True):
    field1: str
    field2: str

class Model(BaseModel):
    data: Table

test = Model(data=Table(field1="test", field2="test"))

a = test.data.field1
# Type of "data" is unknown Pylance(reportUnknownMemberType)
# Type of "field1" is unknown Pylance(reportUnknownMemberType)
# Link: https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnknownMemberType
# (variable) test: Model
```

### Description

1. Create a SQLModel 
2. Create a BaseModel that references the SQLModel
3. MyPy can't resolve the SQLModel type on an instance of the BaseModel

### Operating System

Linux

### SQLModel Version

0.0.16

### Python Version

Python 3.11.8

### Additional Context

<img width="488" alt="Screenshot 2024-02-27 at 12 17 34 AM" src="https://github.com/tiangolo/sqlmodel/assets/14285409/936ef314-a639-4af9-b07b-80a4e23cc867">

**Comment by aistellar** [comment]
I'm having the same issue. It breaks all type hints related to SQLModel classes in VS Code.

**Comment by simonmoesorensen** [comment]
This might also be relevant here. I experience the same issues when dealing with relationships and methods in a `SQLModel`.

```python

class Team(SQLModel):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    heroes: List["Hero"] = Relationship(back_populates="team")

    def team_power(self) -> str:
        if self.heroes:
            powers: List[str] = []
            for hero in self.heroes:
                powers.append(hero.power)  # This works?
        else:
            powers = ["No powers"]

        return f"Team has powers: {powers}"


class Hero(SQLModel):
    id: Optional[int] = Field(default=None, primary_key=True)
    team_id: Optional[int] = Field(default=None, foreign_key="team.id")
    team: Team = Relationship(back_populates="heroes")
    power: str

    def hero_power(self) -> str:
        if self.team is not None:  # Type of "team" is unknown
            team = self.team.name  # Type of "team" is unknown
        else:
            team = "No team"
        return f"Hero is in {team} and has {self.power}!"
```

The strange thing is that it works for `hero.power` in the `Team` class but not the `self.team.name` in the `Hero` class

**Comment by YuriiMotov** [comment]
Works fine in current version (0.0.24)
<img width="444" height="227" alt="image" src="https://github.com/user-attachments/assets/78220921-f359-42a4-a97a-587326ad24b6" />

**Reply by simonmoesorensen** [reply]
Tested it with the Hero/Team example above as well. It works :)
